common/djangoapps/student/models.py

```diff
@@ -1030,32 +1032,37 @@ def get_or_create_enrollment(cls, user, course_key):
         if user.id is None:
             user.save()
 
-        try:
-            enrollment, created = cls.objects.get_or_create(
-                user=user,
-                course_id=course_key,
-            )
+        # Even with MySQL set to use the READ COMMITTED isolation level, get_or_create sometimes
+        # raises an IntegrityError and the object in question doesn't appear in a subsequent get()
+        # call. Paired with non_atomic_requests, the retry logic here is meant to prevent IntegrityErrors
+        # from causing authenticated requests to fail.
```

Comment:
@tasawernawaz please don't copy comments like this without reading them carefully. The edxapp database does not use the READ COMMITTED isolation level. The solution you've copied here will only work if this request is non-atomic, hence the need for [this](https://github.com/edx/programs/pull/138/files#diff-3206b90374ff91e0dda46a06e3e602c6R100), mentioned in the comment. Have you checked if that's necessary here?